test/os-test.cc

```diff
@@ -74,6 +74,24 @@ TEST(os_test, error_code) {
   EXPECT_EQ(error_code(42).get(), 42);
 }
 
+TEST(os_test, format_std_error_code) {
+  EXPECT_EQ("generic:42",
+            fmt::format("{0}", std::error_code(42, std::generic_category())));
```

Comment:
Please wrap the format string in FMT_STRING - this will guarantee that parsing is constexpr and will ensure that the formatter works with format string compilation.